cx_Freeze/freezer.py

```diff
@@ -648,6 +648,14 @@ def _WriteModules(self, fileName, finder):
                         ignore=ignorePatterns,
                     )
 
+                    # remove the subfolders which belong to excluded modules
+                    excludedFolders = [m[len(module.name) + 1:].replace(".", os.sep)
+                                       for m in self.excludeModules if m.startswith(module.name)]
+                    for folder in excludedFolders:
+                        folderToRemove = os.path.join(targetPackageDir, folder)
+                        print("Removing", folderToRemove + "...")
+                        shutil.rmtree(folderToRemove)
+
```

Comment:
Can you use the ignore patterns option of copytree? So you eliminate the copy and delete.
For this case, you should implement a function to extend what actually we are. But this is not a priority. Can be made in another PR if you do not have time.

Reply:
I've actually tried using ignore patterns from copytree, but it won't work as expected. I'll explain why.

The `ignore` parameter from `copytree` expects to receive a callable with two arguments:
* The directory being visited
* The list of contents returned by `os.listdir()` (i.e. bare file and folder names)

Let's have this example:
- Main package name: `my_package`
- Folder structure: 
  - my_package
    - folder_a
       - subfolder_1
       - subfolder_2
       - subfolder_3
       - additional files
- Package to exclude: `my_package.folder_a.subfolder_1`

By default, `copytree` will copy all content under `my_package`. If we would like to use the ignore function, we would need to exclude the whole `folder_a` (as it will be listed by os.listdir) and there's no way of excluding `folder_a/subfolder_1`.

I've spent quite a few time trying different approaches, but the only way I found to make it properly is by deleting the folders once they're copied. I'm open to more ideas if anyone else has a better way.

Reply:
If you join the visited directory with the returned by listdir, then you have a fullname to compare and see that it cannot be copied, so you can wait for the next round, no?

Reply:
No, I don't think so. In that example, we only have the full name when we are visiting `folder_a`. At that point, everything has been copied in the previous visit of `my_package`.

Reply:
Ok. Later I test it and merge. Thanks!

Reply:
Great, thanks! Please let me know if you find any issues. And thanks for supporting this excellent tool :)